plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/services/RepositoryFormService.js

```diff
@@ -0,0 +1,268 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+Ext.define('NX.coreui.services.RepositoryFormService', {
+  singleton: true,
+  mixins: [
+    'NX.LogAware'
+  ],
+  requires: [
+    'NX.I18n'
+  ],
+
+  facets: {
```

Comment:
I did some reading up on ExtJS class system this weekend and am thinking that this simple 3 method 'interface' could be handled by a mixin, separating out the 'facets' per recipe and providing a registry keyed on the recipe name - something similar to the Icon controller. Still not entirely sure how to get then to dynamically register in ExtJS, but I'm sure we'll figure something out :)
How does that sound to you @jdillon ?